binding/form_mapping_test.go

```diff
@@ -288,3 +290,93 @@ func TestMappingIgnoredCircularRef(t *testing.T) {
 	err := mappingByPtr(&s, formSource{}, "form")
 	assert.NoError(t, err)
 }
+
+// this structure has special json unmarshaller, in order to parse email (as an example) as specific structure
+type withJsonUnmarshaller struct {
+	Name string
+	Host string
+}
+
+func (o *withJsonUnmarshaller) UnmarshalJSON(data []byte) error {
+	elems := strings.Split(string(data), "@")
+	if len(elems) != 2 {
+		return errors.New("cannot parse %q as email")
+	}
+	o.Name = elems[0]
+	o.Host = elems[1]
+	return nil
+}
+
+func TestMappingStructFieldJSONUnmarshaller(t *testing.T) {
+	var s struct {
+		Email withJsonUnmarshaller
+	}
+
+	err := mappingByPtr(&s, formSource{"Email": {`test@example.org`}}, "form")
+	assert.NoError(t, err)
+	assert.Equal(t, "test", s.Email.Name)
+	assert.Equal(t, "example.org", s.Email.Host)
+
+	err = mappingByPtr(&s, formSource{"Email": {`not an email`}}, "form")
```

Comment:
Ideally fuzzing would be a separate test. Right now it's not testing 1 unit of code. Also helps with readability.